src/community/features-templating/features-templating-core/src/main/java/org/geoserver/featurestemplating/builders/JSONFieldSupport.java

```diff
@@ -76,14 +91,40 @@ public static boolean isJSONField(PropertyDescriptor opd) {
         return isJSONField(Optional.ofNullable(opd));
     }
 
+    /**
+     * Checks if the given PropertyDescriptor is backed by a JSONB field.
+     *
+     * @param opd PropertyDescriptor to check
+     * @return true if the PropertyDescriptor is backed by a JSONB field
+     */
+    public static boolean isJSONBField(PropertyDescriptor opd) {
+        return isJSONBField(Optional.ofNullable(opd));
+    }
+
     private static boolean isJSONField(Optional<PropertyDescriptor> opd) {
         return opd.map(pd -> pd.getUserData().get(JDBC_NATIVE_TYPENAME))
                 .filter(t -> t instanceof String)
                 .map(t -> (String) t)
-                .filter(t -> "JSON".equalsIgnoreCase(t) || "JSONB".equalsIgnoreCase(t))
+                .filter(t -> matchesJSONType(t, false))
                 .isPresent();
     }
 
+    private static boolean isJSONBField(Optional<PropertyDescriptor> opd) {
+        return opd.map(pd -> pd.getUserData().get(JDBC_NATIVE_TYPENAME))
+                .filter(t -> t instanceof String)
+                .map(t -> (String) t)
+                .filter(t -> matchesJSONType(t, true))
+                .isPresent();
+    }
```

Comment:
``isJSONBField`` returns true even if the native typename is JSON.

However, JSON native type has stable key order,  it's only JSONB that shuffles them.

Reply:
switched to separate method